File: Sorting Algos/quickSort.cpp
#include <iostream>
using namespace std;
int partitionIndex(int *a,int low,int high)
{
    int pivot=a[high]/*element jisse humko correct position par le jana hai*/;
    int i=low-1;
    for(int j=low;j<high;j++)
    {
        if(a[j]<pivot)
        {
            i++;
            swap(a[i],a[j]);
        }
    }
    swap(a[i+1],a[high]);
    return i+1;
}

void quickSort(int *a,int low,int high)
{
    if(low<high)
    {
        ///let pivot = last element of array
        int pi=partitionIndex(a,low,high);
        quickSort(a,low,pi-1);
        quickSort(a,pi+1,high);
    }
}
int main()
{
    int n;
    cout<<"Enter no of elements you want to sort\n";
    cin>>n;
    int *a=new int[n];
    for(int i=0;i<n;i++){
        cin>>a[i];
    }
    cout<<"\nBefore Sorting :\n";
    for(int i=0;i<n;i++){
        cout<<a[i]<<" ";
    }
    quickSort(a,0,n-1);
    cout<<"\nAfter Sorting :\n";
    for(int i=0;i<n;i++){
        cout<<a[i]<<" ";
    }
    return 0;
}

File: Sorting Algos/QuickSort.py
def partition(arr,low,high): 
    i = ( low-1 )     
    pivot = arr[high]  
  
    for j in range(low , high):  
        if arr[j] < pivot:  
            i = i+1 
            arr[i],arr[j] = arr[j],arr[i] 
  
    arr[i+1],arr[high] = arr[high],arr[i+1] 
    return i+1 
  
def quickSort(arr,low,high): 
    if low < high:  
        pi = partition(arr,low,high) 
        quickSort(arr, low, pi-1) 
        quickSort(arr, pi+1, high)

arr = list(map(int,input('Enter the array to be sorted ').split()))

quickSort(arr,0,len(arr)-1)

print('Sorted array')
for i in range(len(arr)):
    print ("% d" % arr[i],end = ' ')